File: 30-09-2025_Generate_all_binary_strings.java


class Solution {
    public void f(String str, List<String> result, int n) {
        if (str.length() == n) {
            result.add(str);
            return;
        }
        f(str + "0", result, n);
        f(str + "1", result, n);
    }

    public ArrayList<String> binstr(int n) {
        ArrayList<String> result = new ArrayList<>();
        f("", result, n);
        return result;
    }
}

File: 30-09-2025_Generate_all_binary_strings.py
class Solution:
    def f(self, s, result, n):
        if len(s) == n:
            result.append(s)
            return
        self.f(s + "0", result, n)
        self.f(s + "1", result, n)

    def binstr(self, n):
        result = []
        self.f("", result, n)
        return result
